wpimath/src/main/native/cpp/jni/WPIMathJNI.cpp

```diff
@@ -43,6 +44,46 @@ bool check_stabilizable(const Eigen::Ref<const Eigen::MatrixXd>& A,
   return true;
 }
 
+std::vector<double> GetElementsFromTrajectory(
+    const frc::Trajectory& trajectory) {
+  std::vector<double> elements;
+  elements.reserve(trajectory.States().size() * 7);
+
+  for (auto&& state : trajectory.States()) {
+    elements.push_back(state.t.to<double>());
+    elements.push_back(state.velocity.to<double>());
+    elements.push_back(state.acceleration.to<double>());
+    elements.push_back(state.pose.X().to<double>());
+    elements.push_back(state.pose.Y().to<double>());
+    elements.push_back(state.pose.Rotation().Radians().to<double>());
+    elements.push_back(state.curvature.to<double>());
+  }
+
+  return elements;
+}
+
+frc::Trajectory CreateTrajectoryFromElements(wpi::ArrayRef<double> elements) {
+  // Make sure that the elements have the correct length.
+  assert(elements.size() % 7 == 0);
+
+  // Create a vector of states from the elements.
+  std::vector<frc::Trajectory::State> states;
+  states.reserve(elements.size() / 7);
+
+  for (size_t i = 0; i < elements.size(); i += 7) {
+    states.emplace_back(frc::Trajectory::State{
```

Comment:
With emplace you don't need to do the constructor call at all, you can just call states.emplace_back() with the constructor arguments directly.

Reply:
There's no explicit constructor for the struct that takes in those arguments so emplace_back won't work like that. We can add it however.

Reply:
Oh, I see.  Probably not worth it for that.